src/main/java/com/epam/aidial/core/AiDial.java

```
@@ -81,8 +92,17 @@ void start() throws Exception {
             }
             EncryptionService encryptionService = new EncryptionService(Json.decodeValue(settings("encryption").toBuffer(), Encryption.class));
             TokenStatsTracker tokenStatsTracker = new TokenStatsTracker();
-            proxy = new Proxy(vertx, client, configStore, logStore, rateLimiter, upstreamBalancer, accessTokenValidator,
-                    storage, encryptionService, apiKeyStore, tokenStatsTracker);
+
+            redis = openRedis();
```

resource service - mandatory core component, we should fail if some configs are missing

---

This way most of the tests will fail. Let's postpone enforcing always redis available.